### ISSUE TITLE
Restrict buffered value size for blob zstd dictionary compression

### ISSUE THREAD

**Issue by Connor1996** [tikv/titan#197]
When blob zstd dictionary compression is enabled, all the values will be buffered and replayed after the compression dictionary is finalized. So if there are multiple concurrent flushes and compactions, the memory footage would be considerable considering the blob file size is 256MB by default. 
For a small RAM instance, it may cause OOM easily. It's better to add a config to control the max concurrent buffered size.

**Comment by yiwu-arbug** [comment]
yes, we saw similar issue when enabling dictionary compression on vanilla RocksDB. we end up limit the sample size to 8MB per SST. cc @hunterlxt

**Comment by ZhenhanGong** [comment]
We can use zstd_max_train_bytes to limit the max buffered size. @Connor1996

**Comment by hunterlxt** [comment]
> We can use zstd_max_train_bytes to limit the max buffered size. @Connor1996

`zstd_max_train_bytes` only affect the bottommost level. And reducing `zstd_max_train_bytes` creates potentially small dictionary, which reduces the compression ration.

**Comment by ZhenhanGong** [comment]
> > We can use zstd_max_train_bytes to limit the max buffered size. @Connor1996
> 
> `zstd_max_train_bytes` only affect the bottommost level. And reducing `zstd_max_train_bytes` creates potentially small dictionary, which reduces the compression ration.

This "zstd_max_train_bytes" belongs to Titan's config, RocksDB has a config with the same name.
Titan's config is used for value compression, while RocksDB's is used for key compression.
I check the code to confirm that this "zstd_max_train_bytes" can be used to limit sampling buffer size.
